From: Michael Froman <mfroman@mozilla.com>
Date: Sat, 17 Jan 2026 15:18:15 -0600
Subject: (tmp-cherry-pick) Revert "Privatize signals in async_packet_socket.h"
 (b2a3c095a5)

This reverts commit f672636abf8b5ff5b04265986ce66d136f272d3a.

Reason for revert: Chrome wasn't quite ready.

Original change's description:
> Privatize signals in async_packet_socket.h
>
> Bug: webrtc:42222066
> Change-Id: I54e47aed3a44e4f6ebdda7ee31d6aef180da966a
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/424120
> Reviewed-by: Tomas Gunnarsson <tommi@webrtc.org>
> Commit-Queue: Harald Alvestrand <hta@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46220}

Bug: webrtc:42222066
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Change-Id: Id12fcc2d30da673f5b9b51d564dbb42fccee2e7f
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/426662
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: Harald Alvestrand <hta@webrtc.org>
Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#46223}
---
 p2p/base/async_stun_tcp_socket_unittest.cc |  9 +++------
 p2p/base/tcp_port.cc                       |  8 ++------
 rtc_base/async_packet_socket.h             | 10 +++++-----
 rtc_base/async_tcp_socket.cc               |  2 +-
 4 files changed, 11 insertions(+), 18 deletions(-)

diff --git a/p2p/base/async_stun_tcp_socket_unittest.cc b/p2p/base/async_stun_tcp_socket_unittest.cc
index 4f3dc9cdb2..c721a47d1b 100644
--- a/p2p/base/async_stun_tcp_socket_unittest.cc
+++ b/p2p/base/async_stun_tcp_socket_unittest.cc
@@ -76,7 +76,7 @@ class AsyncStunServerTCPSocket : public AsyncTcpListenSocket {
                            std::unique_ptr<Socket> socket)
       : AsyncTcpListenSocket(env, std::move(socket)) {}
   void HandleIncomingConnection(std::unique_ptr<Socket> socket) override {
-    NotifyNewConnection(this, new AsyncStunTCPSocket(env(), std::move(socket)));
+    SignalNewConnection(this, new AsyncStunTCPSocket(env(), std::move(socket)));
   }
 };
 
@@ -95,11 +95,8 @@ class AsyncStunTCPSocketTest : public ::testing::Test,
     server->Bind(kServerAddr);
     listen_socket_ =
         std::make_unique<AsyncStunServerTCPSocket>(env, std::move(server));
-    listen_socket_->SubscribeNewConnection(
-        this, [this](AsyncListenSocket* listen_socket,
-                     AsyncPacketSocket* packet_socket) {
-          OnNewConnection(listen_socket, packet_socket);
-        });
+    listen_socket_->SignalNewConnection.connect(
+        this, &AsyncStunTCPSocketTest::OnNewConnection);
 
     std::unique_ptr<Socket> client =
         vss_->Create(kClientAddr.family(), SOCK_STREAM);
diff --git a/p2p/base/tcp_port.cc b/p2p/base/tcp_port.cc
index 87ef5bffd0..aa297448c7 100644
--- a/p2p/base/tcp_port.cc
+++ b/p2p/base/tcp_port.cc
@@ -310,11 +310,7 @@ void TCPPort::TryCreateServerSocket() {
         << ": TCP server socket creation failed; continuing anyway.";
     return;
   }
-  listen_socket_->SubscribeNewConnection(
-      this, [this](AsyncListenSocket* listen_socket,
-                   AsyncPacketSocket* packet_socket) {
-        OnNewConnection(listen_socket, packet_socket);
-      });
+  listen_socket_->SignalNewConnection.connect(this, &TCPPort::OnNewConnection);
 }
 
 AsyncPacketSocket* TCPPort::GetIncoming(const SocketAddress& addr,
@@ -339,7 +335,7 @@ void TCPPort::OnReadPacket(AsyncPacketSocket* socket,
 
 void TCPPort::OnSentPacket(AsyncPacketSocket* socket,
                            const SentPacketInfo& sent_packet) {
-  NotifySentPacket(sent_packet);
+  PortInterface::SignalSentPacket(sent_packet);
 }
 
 void TCPPort::OnReadyToSend(AsyncPacketSocket* socket) {
diff --git a/rtc_base/async_packet_socket.h b/rtc_base/async_packet_socket.h
index ee66e76238..1bcfa4f30c 100644
--- a/rtc_base/async_packet_socket.h
+++ b/rtc_base/async_packet_socket.h
@@ -142,6 +142,7 @@ class RTC_EXPORT AsyncPacketSocket {
   void DeregisterReceivedPacketCallback();
 
   // Emitted each time a packet is sent.
+  sigslot::signal2<AsyncPacketSocket*, const SentPacketInfo&> SignalSentPacket;
   void SubscribeSentPacket(
       void* tag,
       absl::AnyInvocable<void(AsyncPacketSocket*, const SentPacketInfo&)>
@@ -154,6 +155,7 @@ class RTC_EXPORT AsyncPacketSocket {
   }
 
   // Emitted when the socket is currently able to send.
+  sigslot::signal1<AsyncPacketSocket*> SignalReadyToSend;
   void SubscribeReadyToSend(
       void* tag,
       absl::AnyInvocable<void(AsyncPacketSocket*)> callback) {
@@ -169,6 +171,7 @@ class RTC_EXPORT AsyncPacketSocket {
   // Emitted after address for the socket is allocated, i.e. binding
   // is finished. State of the socket is changed from BINDING to BOUND
   // (for UDP sockets).
+  sigslot::signal2<AsyncPacketSocket*, const SocketAddress&> SignalAddressReady;
   void SubscribeAddressReady(
       void* tag,
       absl::AnyInvocable<void(AsyncPacketSocket*, const SocketAddress&)>
@@ -185,6 +188,7 @@ class RTC_EXPORT AsyncPacketSocket {
 
   // Emitted for client TCP sockets when state is changed from
   // CONNECTING to CONNECTED.
+  sigslot::signal1<AsyncPacketSocket*> SignalConnect;
   void NotifyConnect(AsyncPacketSocket* socket) { SignalConnect(socket); }
   void SubscribeConnect(absl::AnyInvocable<void(AsyncPacketSocket*)> callback) {
     connect_trampoline_.Subscribe(std::move(callback));
@@ -219,16 +223,12 @@ class RTC_EXPORT AsyncPacketSocket {
       RTC_GUARDED_BY(&network_checker_);
   absl::AnyInvocable<void(AsyncPacketSocket*, const ReceivedIpPacket&)>
       received_packet_callback_ RTC_GUARDED_BY(&network_checker_);
-  sigslot::signal1<AsyncPacketSocket*> SignalConnect;
   SignalTrampoline<AsyncPacketSocket, &AsyncPacketSocket::SignalConnect>
       connect_trampoline_;
-  sigslot::signal2<AsyncPacketSocket*, const SentPacketInfo&> SignalSentPacket;
   SignalTrampoline<AsyncPacketSocket, &AsyncPacketSocket::SignalSentPacket>
       sent_packet_trampoline_;
-  sigslot::signal1<AsyncPacketSocket*> SignalReadyToSend;
   SignalTrampoline<AsyncPacketSocket, &AsyncPacketSocket::SignalReadyToSend>
       ready_to_send_trampoline_;
-  sigslot::signal2<AsyncPacketSocket*, const SocketAddress&> SignalAddressReady;
   SignalTrampoline<AsyncPacketSocket, &AsyncPacketSocket::SignalAddressReady>
       address_ready_trampoline_;
 };
@@ -251,6 +251,7 @@ class RTC_EXPORT AsyncListenSocket {
   // socket is not bound yet (GetState() returns kBinding).
   virtual SocketAddress GetLocalAddress() const = 0;
 
+  sigslot::signal2<AsyncListenSocket*, AsyncPacketSocket*> SignalNewConnection;
   void SubscribeNewConnection(
       void* tag,
       absl::AnyInvocable<void(AsyncListenSocket*, AsyncPacketSocket*)>
@@ -266,7 +267,6 @@ class RTC_EXPORT AsyncListenSocket {
   }
 
  private:
-  sigslot::signal2<AsyncListenSocket*, AsyncPacketSocket*> SignalNewConnection;
   SignalTrampoline<AsyncListenSocket, &AsyncListenSocket::SignalNewConnection>
       new_connection_trampoline_;
 };
diff --git a/rtc_base/async_tcp_socket.cc b/rtc_base/async_tcp_socket.cc
index d42ccb42b2..001387f47e 100644
--- a/rtc_base/async_tcp_socket.cc
+++ b/rtc_base/async_tcp_socket.cc
@@ -339,7 +339,7 @@ void AsyncTcpListenSocket::OnReadEvent(Socket* socket) {
 
 void AsyncTcpListenSocket::HandleIncomingConnection(
     std::unique_ptr<Socket> socket) {
-  NotifyNewConnection(this, new AsyncTCPSocket(env_, std::move(socket)));
+  SignalNewConnection(this, new AsyncTCPSocket(env_, std::move(socket)));
 }
 
 }  // namespace webrtc
